### PR TITLE
Allow for multiple predicates

### PR DESCRIPTION
**Pull request by vrinek**
I have the need to use multiple predicates on the same method (`HttpClient.post()` called with different arguments).

## TODO

- [x] Write tests
- [x] Update docs

### CODE REVIEW

File: lib/deride.js [hunk 1]
```diff
@@ -284,8 +284,13 @@ function Setup(obj, method, emitter) {
...
         }
-        if (_.isFunction(callBasedOnPredicate)) {
-            return callBasedOnPredicate.apply(self, arguments);
+
+        var args = arguments;
+        var matchingPredicateTuple = functionPredicates.find(function(tuple){
+            return tuple[0].apply(self, args);
+        });
+        if (matchingPredicateTuple) {
+            return matchingPredicateTuple[1].apply(self, arguments);
```
Comment (REAANDREW) [inline lib/deride.js]: Took me a few minutes to understand why you were indexing with a magic number of 1 here until I noticed line 395 where you are using an array as a tuple.  Never done that before in javascript but I like the idea - nice work.

**Comment by REAANDREW** [comment]
Nice work!  Once the docs are updated I will merge it in! :-)

**Comment by vrinek** [comment]
Done with the docs as well!

**Comment by REAANDREW** [comment]
Excellent thanks.  One last thing could you change the destination branch to be develop as opposed to master.  Happy to accept once you have done that!

**Comment by vrinek** [comment]
Changed the base branch!

**Comment by nningego** [comment]
This looks great! When can I expect this to be published to npm? Thanks.

**Comment by jamlen** [comment]
I'm merging to `master` now and will be published in the next 30 mins or so... all being well!

**Comment by jamlen** [comment]
Just published v1.1.0 with this in it. Need to add a `Contributors` file to give credit :)

**Comment by jamlen** [comment]
Oh and you can join the slack channel by signing up here https://guzzlerio.slack.com/signup

**Comment by gbhasha** [comment]
@vrinek Nice Work. Thanks for the contribution.

**Comment by vrinek** [comment]
Great! Thanks!

**Pull request merged**